app/models/report.rb

```diff
@@ -1435,7 +1445,14 @@ def self.report_storage_stats(report)
 
   def self.report_top_uploads(report)
     report.modes = [:table]
-
+    report.filter_options = [
+      {
+        id: "file-extension",
+        selected: report.filter_values.fetch("file-extension", "any"),
+        choices: ["jpeg", "gif", "png"],
+        allowAny: true
```

Comment:
I wonder three things here :p

- do we need a `allowAll` ? as in "any file extension"
- should `allowAny` means you can enter your own value? Maybe we need a better name ?
- for this specific case, could choices use the allowed extensions file setting?

Reply:
1. Yes, I think we do. That being said, I should use: `SiteSetting.authorized_extensions` for `choices` value
2. The assumption now is that it has a default value that is [fetched from client.en.yml](https://github.com/discourse/discourse/pull/6816/files#diff-7ac2736614f0a5678e71cecf1563384eR2867), the default value is: `any` ... I would say this is okay.
3. Yes, totally 😃

Reply:
For me any would mean I could type: "something" and it would use it the where query.

So basically I don’t think this report should support "any". "all" or a choice in the list should be what we need. "any" should also not be a selectable entry in the list but we should only implement combo-box filtering handling.